### ISSUE TITLE
cw_multi_test: make fields pub

### ISSUE THREAD

**Issue by dakom** [CosmWasm/cw-plus#754]
The current implementation makes two restrictive decisions:

1. The only way to create a wrapper is to pass function pointers (can't pass closures or any ol `impl Fn`)
2. All the fields are private

The internal `new()` creator is able to create a closure from the passed-in function pointer, but the current setup disallows the caller to do this too. e.g. it may want to wrap functions _before_ passing them in.

Suggestion: either change the `new()` functions to accept _any_ `impl Fn`, which is going to mean a bunch more generics and probably a bit painful to implement quickly, or just make all the fields `pub`

Since this is purely a testing framework, I'd say just make all the fields `pub` and if anyone wants to manually create a `ContractWrapper` they can do so however they want.

**Comment by hashedone** [comment]
The problem is that implementation if `ContractWrapper` is not trivial, it would require it to be generic over passed handlers types (right now it is generic over magic internals). I think it is doable, I will take a look at it next week, but personally I would suggest just implementing your custom type implementing `Contract` instead of using `ContractWrapper`.

**Comment by dakom** [comment]
Yup! totally agree @hashedone - looks like it would require adding more generics everywhere, which would be painful on the library side (but would make the API nicer for users)

Also agree, implementing our own custom type is totally workable, the issue here should be thought of more as improving developer experience rather than actually blocking anything important.

**Comment by ethanfrey** [comment]
I would put a deep refactor of multi-test before this should be approached (or fix this as part of a refactor). We may be able to simplify the traits and generics if some big brains spend several days focused on it.
